macaroonbakery/checkers/time.py

```diff
@@ -1,18 +1,68 @@
 # Copyright 2017 Canonical Ltd.
 # Licensed under the LGPLv3, see LICENCE file for details.
+
+import pyrfc3339
+
 from macaroonbakery.checkers.auth_context import ContextKey
+from macaroonbakery.checkers.conditions import COND_TIME_BEFORE, STD_NAMESPACE
+from macaroonbakery.checkers.utils import condition_with_prefix
+from macaroonbakery.checkers.caveat import parse_caveat
 
 
 TIME_KEY = ContextKey('time-key')
 
 
 def context_with_clock(ctx, clock):
-    ''' Returns a copy of ctx with a key added that associates it with the given
-    clock implementation, which will be used by the time-before checker
+    ''' Returns a copy of ctx with a key added that associates it with the
+    given clock implementation, which will be used by the time-before checker
     to determine the current time.
     The clock should have a utcnow method that returns the current time
     as a datetime value in UTC.
     '''
     if clock is None:
         return ctx
     return ctx.with_value(TIME_KEY, clock)
+
+
+def macaroons_expiry_time(ns, ms):
+    ''' Returns the minimum time of any time-before caveats found in the given
+    macaroons or None if no such caveats were found.
+    :param ns: a Namespace, used to resolve caveats.
+    :param ms: a list of pymacaroons.Macaroon
+    :return: datetime.DateTime or None.
+    '''
+    t = None
+    for m in ms:
+        et = expiry_time(ns, m.caveats)
+        if et is not None:
```

Comment:
you could collapse the two ifs together to remove one indentation level, eg

if et is not None and (t is None or et < t):

Reply:
done